web-apps/NextJs-App/README.md

```diff
@@ -0,0 +1,231 @@
+
+#### Step 2.3.2: Deploy the web application component
+
+Once you create the web application component, you can deploy it to the Choreo runtime. To deploy the web application component, follow the steps below:
+
+1. In the left menu, click **Deploy**.
+2. In the **Build Area** card, select **Configure & Deploy** from the split button and click to add runtime configurations for your application and deploy. 
+3. Add configurations and deploy. The deployment may take a few minutes to complete. See how to define and read configurations at [Develop a Web Application page](https://wso2.com/choreo/docs/develop-components/develop-a-web-application/#creating-a-web-application).
+4. Once you deploy the web application, copy the **Web App URL** from the development environment card.
+5. Navigate to the web app URL. You can verify that you have successfully hosted the web application.
+
+Although you hosted the web application, you have not configured the web application to securely invoke the service. Let's create an OAuth app in the IdP (Asgardeo) and configure the web app.
+
+#### Step 2.3.3: Configure Asgardeo (IdP) to integrate with your application
+
+Choreo uses Asgardeo as the default identity provider for Choreo applications.
+
+1. Access Asgardeo at [https://console.asgardeo.io/](https://console.asgardeo.io/) and sign in with the same credentials with which you signed in to Choreo.
+2. In the Asgardeo Console's left navigation, click **Application**. Click **New Application** then choose **Standard based application**.
+3. Provide a name for the application. For example, `todoListApp`, choose **OpenIDConnect** and Register.
+3. Click the **Protocol** tab and apply the following changes:
+
+    1. Under **Allowed grant types**, select **Code**.
+    2. Select the **Public client** checkbox.
+    3. In the **Authorized redirect URLs** field, enter {WebApp URL}/api/auth/callback/asgardeo. E.g http://localhost:3000/api/auth/callback/asgardeo`` and click the **+** icon to add the entry.
+    4. In the **Allowed origins** field, add the Web Application URL.
+    6. Under **Access Token**, select **JWT** as the **Token type**.
+    7. Click **Update**.
+
+#### Step 2.3.4: Configure & Deploy the web application component
+
+Once the web application component is created, you can deploy it to the Choreo runtime. To deploy the web application component, follow the steps below:
+
+1. In the left menu, click **Deploy**.
+2. In the **Build Area** card, select **Configure & Deploy** from the split button and click.
+3. In the **Configure & Deploy** panel, add the mount path as /app/.env and add the following configurations in the editor:
+```
+NEXTAUTH_URL=http://localhost:3000
+SECRET={a random secret}
+NEXTAUTH_SECRET={a random secret}
```

Comment:
Shall we add a link which explains what this config is
https://next-auth.js.org/configuration/options